RELEASE.md

```diff
@@ -0,0 +1,20 @@
+# Release
+
+To release new version of Kuma follow the steps:
+
+1. Create a CHANGELOG.md.
+2. Double-check with changelog that all new features are documented on kuma.io website.
+3. Create PR to [kuma.io website repository](https://github.com/Kong/kuma-website) with new download links and the new version.
+4. Create a new git tag.
+5. Push git tag. This will trigger the release job on CI.
```

Comment:
Ideally the release job CI should do step 7 for us ( double check that it works with demo app ). Not necessary at this time but a good CI/CD backlog item to keep in mind